src/accel/SharedParams.hh

```diff
@@ -87,13 +88,20 @@ class SharedParams
 
     using SPHitManager = std::shared_ptr<detail::HitManager>;
     using SPOffloadWriter = std::shared_ptr<detail::OffloadWriter>;
+    using SPRootOffloadWriter = std::shared_ptr<detail::RootOffloadWriter>;
 
     //! Hit manager, to be used only by LocalTransporter
     SPHitManager const& hit_manager() const { return hit_manager_; }
 
     //! Optional offload writer, only for use by LocalTransporter
     SPOffloadWriter const& offload_writer() const { return offload_writer_; }
 
+    //! Optional ROOT offload writer, only for use by LocalTransporter
+    SPRootOffloadWriter const& root_offload_writer() const
+    {
+        return root_offload_writer_;
+    }
+
```

Comment:
Delete all this code; we should be using the same offload writer interface: either HepMC3 or ROOT, we don't need separate copies of both. I just wanted you to add the `RootEvent{Reader,Writer}` class and I was going to add in a separate PR the extra bit of `std::functional` and "make a different writer based on the output extension" code needed to hook it in.